docs/dev/new_check_howto.md

```diff
@@ -88,187 +91,166 @@ A Check is a Python class with the following requirements:
 
 The code would look something like this:
 
 ```python
 import requests
 
-from datadog_checks.checks import AgentCheck
-from datadog_checks.errors import CheckException
+from datadog_checks.base import AgentCheck, ConfigurationError
 
 
-# MyCheck derives from AgentCheck, and provides the required check method.
-class MyCheck(AgentCheck):
+class AwesomeCheck(AgentCheck):
+    """AwesomeCheck derives from AgentCheck, and provides the required check method."""
+
     def check(self, instance):
         url = instance.get('url')
         search_string = instance.get('search_string')
 
-        # It's a good idea to do some basic sanity checking. Try to be as
-        # specific as possible, with the exceptions; you can fall back to
-        # CheckException when in doubt though.
+        # It's a very good idea to do some basic sanity checking.
+        # Try to be as specific as possible with the exceptions.
         if not url or not search_string:
-            raise CheckException("Configuration error, please fix my_check.yaml")
+            raise ConfigurationError('Configuration error, please fix awesome.yaml')
 
         try:
-            r = requests.get(url)
-            r.raise_for_status()
-            if search_string in r.text:
-                # Page is accessible and the string is present.
-                self.service_check('my_check.all_good', self.OK)
-            else:
-                # Page is accessible but the string was not found.
-                self.service_check('my_check.all_good', self.WARNING)
+            response = requests.get(url)
+            response.raise_for_status()
+
+        # Something went horribly wrong
         except Exception as e:
-            # Something went horribly wrong. Ideally we'd be more specific...
-            self.service_check('my_check.all_good', self.CRITICAL, e)
+            # Ideally we'd use a more specific message...
+            self.service_check('awesome.search', self.CRITICAL, message=str(e))
+
+        # Page is accessible
+        else:
+
+            # search_string is present
+            if search_string in response.text:
+                self.service_check('awesome.search', self.OK)
+
+            # search_string was not found
+            else:
+                self.service_check('awesome.search', self.WARNING)
 ```
 
-To learn more about the base Python class, see the [Python API documentation][5]. Moving along, let's dive into tests, which are an important part of any project (and *required* for inclusion in `integrations-extras`).
+To learn more about the base Python class, see the [Python API documentation][5]. Moving along, let's dive into tests, which are an important part of any project (and _required_ for inclusion in `integrations-extras`).
```

Comment:
```suggestion
To learn more about the base Python class, see the [Python API documentation][5].
```

I added this information below just in case someone skips to this section - it is important info for why you need to add tests

Reply:
👍 @apigirl's suggestion